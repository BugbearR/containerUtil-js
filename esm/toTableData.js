/**
 * Convert to table data.
 * @param {Object[]} rows
 * @param {Object} opts
 * @param {String[]} opts.pick include field names
 * @param {String[]} opts.omit exclude field names
 * @returns {Object} table data
 */
export default function toTableData(rows, opts) {
    if (rows === undefined || rows === null) {
        return rows;
    }
    if (!Array.isArray(rows)) {
        return undefined;
    }

    let fields = undefined;
    if (opts && opts.pick && Array.isArray(opts.pick)) {
        fields = opts.pick;
    }
    else {
        if (rows.length === 0) {
            return { fields: [], rows: [] };
        }
        fields = Object.getOwnPropertyNames(rows[0]);
    }

    if (opts && opts.omit && Array.isArray(opts.omit)) {
        const omitSet = new Set(opts.omit);
        fields = fields.filter((field) => {
            return !omitSet.has(field);
        });
    }

    const newRows = rows.map((row) => {
        return fields.map((field) => {
            return row[field];
        });
    });

    return {
        fields: fields,
        rows: newRows
    };
}
