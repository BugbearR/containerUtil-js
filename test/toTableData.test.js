import toTableData from "../esm/toTableData";

test('Hello, world!', () => {
    const data = [
        { ka: "va1", kb: "vb1" },
        { ka: "va2", kb: "vb2" },
    ];
    expect(toTableData(data)).toMatchSnapshot();
});
