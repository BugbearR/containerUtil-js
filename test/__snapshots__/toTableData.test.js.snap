// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello, world! 1`] = `
Object {
  "fields": Array [
    "ka",
    "kb",
  ],
  "rows": Array [
    Array [
      "va1",
      "vb1",
    ],
    Array [
      "va2",
      "vb2",
    ],
  ],
}
`;
